app/src/main/java/net/squanchy/settings/SettingsFragment.kt

```diff
@@ -0,0 +1,132 @@
+import android.support.v7.app.AppCompatActivity
+import android.view.View
+import android.widget.ListView
+
+import com.google.firebase.auth.FirebaseUser
+
+import net.squanchy.BuildConfig
+import net.squanchy.R
+import net.squanchy.navigation.Navigator
+import net.squanchy.signin.SignInService
+import net.squanchy.support.lang.Optional
+
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+
+class SettingsFragment : PreferenceFragment() {
+
+    private val subscriptions = CompositeDisposable()
+
+    private lateinit var signInService: SignInService
+    private lateinit var navigator: Navigator
+
+    private lateinit var accountCategory: PreferenceCategory
+    private lateinit var accountEmailPreference: Preference
+    private lateinit var accountSignInSignOutPreference: Preference
+
+    private val viewOrThrow: View
+        get() = this.view ?: throw IllegalStateException("You cannot access the fragment's view when it doesn't exist yet")
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        addPreferencesFromResource(R.xml.settings_preferences)
+        preferenceScreen.isOrderingAsAdded = false
+
+        displayBuildVersion()
+
+        if (!BuildConfig.DEBUG) {
+            removeDebugCategory()
+        }
+
+        val activity = activity as AppCompatActivity // TODO UNYOLO
+        val component = settingsFragmentComponent(activity)
+
+        signInService = component.signInService()
+        navigator = component.navigator()
+
+        accountCategory = findPreference(getString(R.string.account_category_key)) as PreferenceCategory
+        accountEmailPreference = findPreference(getString(R.string.account_email_preference_key))
+        accountCategory.removePreference(accountEmailPreference)
+        accountSignInSignOutPreference = findPreference(getString(R.string.account_signin_signout_preference_key))
+
+        val aboutPreference = findPreference(getString(R.string.about_preference_key))
+        aboutPreference.setOnPreferenceClickListener {
+            navigator.toAboutSquanchy()
+            true
+        }
+    }
+
+    private fun displayBuildVersion() {
+        val buildVersionKey = getString(R.string.build_version_preference_key)
+        val buildVersionPreference = findPreference(buildVersionKey)
+        val buildVersion = String.format(getString(R.string.version_x), BuildConfig.VERSION_NAME)
+        buildVersionPreference.title = buildVersion
+    }
+
+    private fun removeDebugCategory() {
+        val debugCategoryKey = getString(R.string.debug_category_preference_key)
+        val debugCategory = findPreference(debugCategoryKey)
+        preferenceScreen.removePreference(debugCategory)
+    }
+
+    override fun onStart() {
+        super.onStart()
+
+        hideDividers()
+
+        subscriptions.add(
+                signInService.currentUser()
```

Comment:
Too much indent?

Reply:
No idea why it's indenting like that but with `(` instead of `{` that's what I get. Will check my settings soon-ish